proto/regen/ecocredit/v1/tx.proto

```diff
@@ -215,19 +224,111 @@ message MsgMintBatchCredits {
   string note = 5;
 }
 
-// MsgSealBatch request type for MsgSealBatch RPC.
+// MsgMintBatchCreditsResponse is the Msg/MintBatchCredits response type.
+message MsgMintBatchCreditsResponse {}
+
+// MsgSealBatch request type for SealBatch RPC.
 message MsgSealBatch {
   // Issuer must equal to the batch.issuer address.
   // Signer of the msg.
   string issuer = 1;
 }
 
-// MsgMintBatchCreditsResponse is the Msg/MintBatchCredits response type.
-message MsgMintBatchCreditsResponse {}
-
 // MsgSealBatchResponse is the Msg/SealBatch response type.
 message MsgSealBatchResponse {}
 
+// MsgMintPendingBatchCredits is the Msg/MintPendingBatchCredits request type.
+message MsgMintPendingBatchCredits {
+
+  // issuer is the address of the credit class issuer who is being requested
+  // to approve the batch issuance.
+  string issuer = 1;
+
+  // batch is either an existing batch_denom, or a new set of BatchInfo to be used
+  // for the new batch, if the batch does not exist yet
+  oneof batch {
+    string batch_denom = 2;
+    PendingBatchInfo batch_info = 3;
+  }
+
+  // issuance are the credits proposed by the user to be issued in the batch.
+  repeated BatchIssuance issuance = 4;
+
+  // A reference to a transaction or an event referencing the transaction
+  // which caused the transfer from other chain or registry.
+  // If not set (null), it means that the batch originate in Regen Ledger
+  // (doesn't come from external registry).
+  OriginTx origin_tx = 7;
+
+  // reference note for accounting, will be passed to an event
+  string note = 8;
+
+  message PendingBatchInfo {
+
+    // project is the project_id of the project this batch belongs to, or a
+    // new set of ProjectInfo to be used for the new project, if the project
+    // does not exist yet
+    oneof project {
+      string project_id = 1;
+      PendingProjectInfo project_info = 2;
+    }
+
+    // metadata is any arbitrary metadata attached to the credit batch.
+    string metadata = 3;
+
+    // start_date is the beginning of the period during which this credit batch
+    // was quantified and verified.
+    google.protobuf.Timestamp start_date = 4 [ (gogoproto.stdtime) = true ];
+
+    // end_date is the end of the period during which this credit batch was
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 5 [ (gogoproto.stdtime) = true ];
+
+    // If open is true we will enable future minting.
+    // Otherwise we will seal the batch and disable the future minting.
+    bool open = 6;
+  }
+
+  message PendingProjectInfo {
```

Comment:
To be honest, I don't find the `Info` suffix useful. Why not simply calling it `PendingProject`?

Reply:
We call it `ProjectInfo` and `BatchInfo` in state.proto, so I was just trying to stay consistent with that.

Reply:
I'd tend to agree, although that's already being used for classes, batches and projects: `ClassInfo`, etc.
so we should either keep it or remove it everywhere

Reply:
TBH, I don't see this as a strong argument. If we agree that the style is not good, then let's use a better one than continue going into wrong direction. Especially for naming.